src/classes/DML.cls

```diff
@@ -0,0 +1,53 @@
+/*************************************************************************************************
+* This file is part of the Nebula Framework project, released under the MIT License.             *
+* See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
+*************************************************************************************************/
+public virtual class DML implements IDML {
+    public virtual SObject doInsert(SObject record) { return this.doInsert(new List<SObject>{record})[0];}
```

Comment:
@jamessimone sorry, another question here! I swear we discussed this a month or two ago, but why return SObject (or a list of them) on these methods? The variables passed to the methods should already get updated to have the IDs after the insert occurs, so I don't think there's a need to return anything (plus then you have to cast them from SObject back to the appropriate concrete SObject type).